apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -388,6 +388,11 @@ def write(
                 containing all columns, including the index columns. The schema for the values must
                 match the schema for the :class:`DataFrame`.
 
+                If a column is of categorical type in the schema and a flattened/non-categorical
+                column is presented for data on write, a ``ValueError`` is raised.  If a column is
+                of non-categorical type in the schema and a categorical column is presented for data
+                on write, the data are auto-flattened on behalf of the user.
```

Comment:
I don't think the term `auto-flattened` is a good way to express the operation (the term is used in Pandas to imply index flattening, etc).

Suggestion:

"..., the data are written as an array of category values, and the category type information is not saved."

Or something like that (my suggestion is also not quite right - hoping we can iterate)

Reply:
> hoping we can iterate

Of course!!